Tasks/Common/ServiceFabricHelpers/Get-ServiceFabricEncryptedText.ps1

```diff
@@ -12,7 +12,55 @@ function Get-ServiceFabricEncryptedText
 
     $defaultCertStoreName = "My"
     $defaultCertStoreLocation = "CurrentUser"
-    $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"]
+
+    if ($ClusterConnectionParameters["ServerCertThumbprint"])
+    {
+        $serverCertThumbprints = $ClusterConnectionParameters["ServerCertThumbprint"];
+    }
+    else 
+    {
+        try
+        {
+            $manifest = [xml](Get-ServiceFabricClusterManifest)
+            $securityInfo = $manifest.ClusterManifest.FabricSettings.Section | Where-Object {$_.Name -eq 'Security'}
+            $serverCertsParameter = $securityInfo.Parameter | Where-Object { $_.Name -eq 'ServerCertThumbprints' }
+            $serverCertThumbprints = $serverCertsParameter.Value.Split(',').trim()
```

Comment:
I think it might be possible that $serverCertThumbprints can end up with a null value. Maybe as a safety check the value is value and failover to the common name lookup if it is.

Reply:
Fixed.